File: src/main/java/ru/kainlight/lightshowregion/COMMON/ActionbarManager.java
package ru.kainlight.lightshowregion.COMMON;

import lombok.Getter;
import org.bukkit.entity.Player;
import org.bukkit.scheduler.BukkitScheduler;
import org.bukkit.scheduler.BukkitTask;
import ru.kainlight.lightshowregion.COMMON.lightlibrary.LightPlayer;
import ru.kainlight.lightshowregion.Main;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public final class ActionbarManager {

    private final Main plugin;

    @Getter
    private final Map<Player, BukkitTask> actionbarTask;

    public ActionbarManager(Main plugin) {
        this.plugin = plugin;
        this.actionbarTask = new HashMap<>();
    }

    public void show(Player player) {
        hide(player);
        runTaskUpdateRegionName(player);
    }

    public boolean toggle(Player player) {
        if (hide(player)) {
            return true;
        } else {
            show(player);
            return false;
        }
    }

    public boolean hide(Player player) {
        boolean isShowen = actionbarTask.containsKey(player);
        if (isShowen) {
            actionbarTask.get(player).cancel();
            actionbarTask.remove(player);
        }

        return isShowen;
    }


    private void runTaskUpdateRegionName(Player player) {
        List<String> disabledWorlds = plugin.getConfig().getStringList("region-settings.disabled-worlds");
        String global = plugin.getMessageConfig().getConfig().getString("actionbar.global");

        actionbarTask.put(player, plugin.getServer().getScheduler().runTaskTimerAsynchronously(plugin, () -> {
            if(player == null) return;

            String playerWorldName = player.getWorld().getName();
            if (global == null || global.isEmpty() || disabledWorlds.contains(playerWorldName)) return;

            String regionName = plugin.getRegionManager().sendRegionName(player);
            LightPlayer.of(player).sendActionbar(regionName);
        }, 20L, 20L));
    }

}

File: src/main/java/ru/kainlight/lightshowregion/UTILS/ActionbarManager.kt
package ru.kainlight.lightshowregion.UTILS

import org.bukkit.entity.Player
import org.bukkit.scheduler.BukkitTask
import ru.kainlight.lightlibrary.getAudience
import ru.kainlight.lightlibrary.multiActionbar
import ru.kainlight.lightshowregion.Main

object ActionbarManager {

    val actionbarTask = mutableMapOf<Player, BukkitTask>()

    fun toggle(player: Player): Boolean {
        return if (hide(player)) true else {
            show(player)
            false
        }
    }

    fun show(player: Player) {
        hide(player)

        val isActionbar = Main.getInstance().config.getBoolean("main-settings.actionbar")
        if (isActionbar) runTaskUpdateRegionName(player)
    }

    fun hide(player: Player): Boolean {
        val task = actionbarTask.get(player)

        return if(task == null) false
        else {
            task.cancel()
            true
        }
    }

    fun runTaskUpdateRegionName(player: Player) {
        val disabledWorlds = Main.getInstance().config.getStringList("region-settings.disabled-worlds")
        val global = Main.getInstance().messageConfig.getConfig().getString("actionbar.global")

        actionbarTask.put(player, Main.getInstance().runTaskTimerAsynchronously(Runnable {
            if (!player.isOnline) return@Runnable
            if (global.isNullOrEmpty() || disabledWorlds.contains(player.world.name)) return@Runnable

            val text = RegionManager.getRegionName(player)
            player.getAudience().multiActionbar(text)
        }, 0L, 20L))
    }

}